anta/device.py

```diff
@@ -330,27 +321,53 @@ async def _collect(self, command: AntaCommand) -> None:
         elif self.enable:
             # No password
             commands.append({"cmd": "enable"})
-        if command.revision:
-            commands.append({"cmd": command.command, "revision": command.revision})
-        else:
-            commands.append({"cmd": command.command})
+        commands += [{"cmd": command.command, "revision": command.revision}] if command.revision else [{"cmd": command.command}]
         try:
             response: list[dict[str, Any]] = await self._session.cli(
                 commands=commands,
                 ofmt=command.ofmt,
                 version=command.version,
             )
+            # Does not keep response of 'enable' command
+            command.output = response[-1]
         except aioeapi.EapiCommandError as e:
+            # This block catch exceptions related to EOS issuing an error
             command.errors = e.errors
-            if self.supports(command):
-                logger.error("Command '%s' failed on %s", command.command, self.name)
-        except (HTTPError, ConnectError) as e:
-            command.errors = [str(e)]
-            logger.error("Cannot connect to device %s", self.name)
-        else:
-            # selecting only our command output
-            command.output = response[-1]
-            logger.debug("%s: %s", self.name, command)
+            if command.requires_privileges:
+                logger.error(
+                    "Command '%s' requires privileged mode on %s. Verify user permissions and if the `enable` option is required.", command.command, self.name
+                )
+            if command.supported:
+                logger.error("Command '%s' failed on %s: %s", command.command, self.name, e.errors[0] if len(e.errors) == 1 else e.errors)
+            else:
+                logger.debug("Command '%s' is not supported on '%s' (%s)", command.command, self.name, self.hw_model)
+        except TimeoutException as e:
+            # This block catch exceptions related timeouts
+            command.errors = [exc_to_str(e)]
+            timeouts = self._session.timeout.as_dict()
+            logger.error(
+                "%s occurred while sending a command to %s. Consider increasing the timeout.\nCurrent timeouts: Connect: %s | Read: %s | Write: %s | Pool: %s",
+                exc_to_str(e),
+                self.name,
+                timeouts["connect"],
+                timeouts["read"],
+                timeouts["write"],
+                timeouts["pool"],
+            )
+        except (ConnectError, OSError) as e:
+            # This block catch exceptions related to OSError and sockets issues
+            command.errors = [exc_to_str(e)]
+            if (isinstance(exc := e.__cause__, httpcore.ConnectError) and isinstance(os_error := exc.__context__, OSError)) or isinstance(os_error := e, OSError):  # pylint: disable=no-member
+                if isinstance(os_error.__cause__, OSError):
```

Comment:
Hmm so in both cases `os_error` variable is an OSError, ar eyou suggesting it can contains a nested OSError in __cause__ ?

Reply:
Yes
The OS can refuse to open a file descriptor at any layer of the stack httpx->hhtpcore->anyio somehting like than.
In some cases, the OSError is located in the cause of the cause of the caught exception, in other it is just the cause.